common/changes/@rushstack/eslint-config/user-danade-BumpEslintUtils_2024-08-13-00-25.json

```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@rushstack/eslint-config",
+      "comment": "Bump \"@typescript-eslint/eslint-plugin\" to \"~8.1.0\" and \"@typescript-eslint/eslint-parser\" to \"~8.1.0\". Note that this means that eslint@^8.57.0 is now required due to breaking changes in rules and parser.",
+      "type": "major"
```

Comment:
Major? Guess that's okay. @octogonz - objections?

Reply:
In my experience, forcing people to upgrade to a new _minor_ version of ESLint has a fairly low risk of breaking anything.

However bumping two _major_ versions of `@typescript-eslint` is very likely to break things, so I agree with a MAJOR version bump.  It would be wise to take a look at the `@typescript-eslint` changelogs to understand what might have been broken, since "it compiles" is not a guarantee that nothing was regressed.  It's also possible they dropped support for older TypeScript compiler versions or Node.js, in which case we might want to mention that as well.

It would also be great to validate this PR by running against a large monorepo besides Rush Stack. (Fine if we do it after publishing, just it would be better for us to catch it than someone having to report an issue.)

Reply:
@octogonz that is the plan. I want to pull this into our monorepo ASAP after publishing, along with a couple other changes I plan to get out today. I'll update the change message to explicitly note the breaks for both Node 16 and parser versions < 8.57.0.